c/tskit/tables.h

```diff
@@ -2446,6 +2445,27 @@ nodes (and individuals and populations) reordered.
 int tsk_table_collection_subset(
     tsk_table_collection_t *self, tsk_id_t *nodes, tsk_size_t num_nodes);
 
+/**
+@brief Forms the node-wise union of two table collections.
+
+@rst
+Expands the table collection by forming the union of itself with ``other``.
+
+.. note:: Migrations are currently not supported by union, and an error will
+    be raised if we attempt call union on a table collection with migrations.
+@endrst
+
+@param self A pointer to a tsk_table_collection_t object.
+@param other A pointer to a tsk_table_collection_t object.
+@param nodes An array of length other.num_nodes in which the indexes represents
+node IDs in other and the values represent the equivalent node in self or
+TSK_NULL if unique to other.
```

Comment:
How about
```
An array of node IDs that related nodes in other to nodes in self: the k-th element of nodes should be the index of the equivalent node is self, or TSK_NULL if the node is not present in self (in which case it will be added to self).
```